### ISSUE TITLE
Terminal cannot be navigated via keyboard

### ISSUE THREAD

**Issue by IsabelParedes** [jupyterlab/jupyterlab#18704]
## Description

Currently when opens a new terminal and the terminal window receives focus, it is not possible to navigate outside of the terminal using only the keyboard. The terminal is currently a tab trap.

![Image](https://github.com/user-attachments/assets/ac672f25-880d-4fb4-bdb8-cfcd6b621e4f)

## Reproduce

1. Open a new terminal
2. Press `Esc` or `Tab` to attempt to navigate out of the terminal

## Expected behavior

- Pressing `Esc` when the terminal is in focus should move the focus to the terminal widget.
- Afterwards, pressing `Tab` should move focus to the next element and not the terminal input area.

## Context

- Operating System and version: Arch Linux
- Browser and version: Chromium v146.0.7680.153
- JupyterLab version: 4.5.5
